File: backend/src/main/java/ResumeBuilder/backend/Models/DatabaseModels/UserAccountModel.java
package ResumeBuilder.backend.Models.DatabaseModels;

import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "user_account")
public class UserAccountModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    public int id;

    @Column(unique = true)
    public String username;

    @Column
    public String password;

    @Column(nullable = false)
    public boolean admin;

    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "user_info_id", nullable = false)
    public UserInfoModel userInfoModel;
}

File: frontend/models/UserAccountModel.js
import UserInfoModel from "./UserInfoModel";
import { v4 as uuid } from "uuid";

export default class UserAccountModel {
  id = uuid();
  username = "";
  password = "";
  userInfoModel = new UserInfoModel();

  constructor(data) {
    if (data != null) {
      this.id = data.id;
      this.username = data.username;
      this.password = data.password;
      this.userInfoModel = data.userInfoModel;
    }
  }
}
